client/collection.go

```diff
@@ -136,6 +136,18 @@ type Collection interface {
 
 	// GetAllDocKeys returns all the document keys that exist in the collection.
 	GetAllDocKeys(ctx context.Context) (<-chan DocKeysResult, error)
+
+	// CreateIndex creates a new index on the collection.
+	CreateIndex(ctx context.Context, desc IndexDescription) (IndexDescription, error)
+
+	// DropIndex drops an index from the collection.
+	DropIndex(ctx context.Context, indexName string) error
+
+	// DropAllIndexes drops all indexes from the collection.
+	DropAllIndexes(ctx context.Context) error
```

Comment:
thought: This looks like a fun function, I would suggest we do not offer this level of destruction to our users, I think dropping of indexes should be deliberate.

Reply:
I though this would be a convenient method for collection deletion so that we don't need to go through additional hassle of retrieving the list of current indexes and delete them one by one.

Reply:
That's a good idea although we could probably achieve this using a method on collection without it being part of the interface.

Reply:
made it private